Sources/ProjectDescription/Settings.swift

```diff
@@ -21,18 +21,34 @@ public class Configuration: Codable {
     }
 }
 
+// MARK: - DefaultSettings
+
+/// Specifies the default set of settings applied to all the projects and targets.
+/// The default settings can be overridden via `Settings base: [String: String]`
+/// and `Configuration settings: [String: String]`.
+///
+/// - all: Essential settings plus all the recommended settings (including extra warnings)
+/// - essential: Only essential settings to make the projects compile (i.e. `TARGETED_DEVICE_FAMILY`)
+public enum DefaultSettings: String, Codable {
```

Comment:
I'd add a new section to the manifest documentation page like [here](https://github.com/tuist/tuist/pull/380#discussion_r288534250).